cmd/config-manager/main.go

```diff
@@ -29,33 +29,79 @@ import (
 )
 
 const (
-	tomlFilePath = "/etc/containerd/config.toml"
-	nriPluginKey = "io.containerd.nri.v1.nri"
-	disableKey   = "disable"
-	replaceMode  = "replace"
-	resultDone   = "done"
-	unit         = "containerd.service"
+	containerdConfigFile = "/etc/containerd/config.toml"
+	crioConfigFile       = "/etc/crio/crio.conf.d/10-enable-nri.conf"
+	nriPluginKey         = "io.containerd.nri.v1.nri"
+	replaceMode          = "replace"
+	resultDone           = "done"
+	containerdUnit       = "containerd.service"
+	crioUnit             = "crio.service"
 )
 
 func main() {
-	tomlMap, err := readConfig(tomlFilePath)
+	unit, err := detectRuntime()
 	if err != nil {
-		log.Fatalf("Error reading TOML file: %v", err)
+		log.Fatalf("failed to autodetect container runtime: %v", err)
 	}
 
-	updatedTomlMap := updateNRIPlugin(tomlMap)
+	switch unit {
+	case containerdUnit:
+		err = enableNriForContainerd()
+	case crioUnit:
+		err = enableNriForCrio()
+	default:
+		log.Fatalf("unknown container runtime %q", unit)
+	}
+
+	if err != nil {
+		log.Fatalf("error enabling NRI: %v", err)
+	}
+
+	if err = restartSystemdUnit(unit); err != nil {
+		log.Fatalf("failed to restart %q unit: %v", unit, err)
+	}
+
+	log.Println("enabled NRI for", unit)
+}
 
-	err = writeConfig(tomlFilePath, updatedTomlMap)
+func enableNriForContainerd() error {
+	tomlMap, err := readConfig(containerdConfigFile)
 	if err != nil {
-		log.Fatalf("failed to write updated config into a file %q:, %v", tomlFilePath, err)
+		return fmt.Errorf("error reading TOML file: %v", err)
 	}
 
-	err = restartSystemdUnit(unit)
+	updatedTomlMap := updateContainerdConfig(tomlMap)
+
+	err = writeToContainerdConfig(containerdConfigFile, updatedTomlMap)
 	if err != nil {
-		log.Fatalf("failed to restart containerd: %v", err)
+		return fmt.Errorf("failed to write updated config into a file %q:, %v", containerdConfigFile, err)
```

Comment:
nit: unintentional '%q:, %v', guessing was meant to be '%q: %v'

Reply:
thanks, fixed